Updated DEV guide with the new initialization api (#820)
opensearch-project/opensearch-sdk-java: DEVELOPER_GUIDE.md

- Start a separate terminal and navigate to the directory where OpenSearch has been cloned using `cd OpenSearch`.
- Run `./gradlew assemble` to create a local distribution.
- Start OpenSearch using `./bin/opensearch`.

> - Hit the below sample REST API to initialize an extension

Reviewer: ```suggestion
- Query the below sample REST API to initialize an extension
```

```

>  curl -XPOST "localhost:9200/_extensions/initialize" -H "Content-Type:application/json" --data '{"name":"ad-extension","uniqueId":"ad-extension","hostAddress":"127.0.0.1","port":"4532","version":"1.0","opensearchVersion":"3.0.0","minimumCompatibleVersion":"3.0.0","dependencies":[{"uniqueId":"test1","version":"2.0.0"},{"uniqueId":"test2","version":"3.0.0"}]}'

Reviewer: This scrolls off the screen in markdown.  

![Screenshot 2023-06-12 at 2 24 21 PM](https://github.com/opensearch-project/opensearch-sdk-java/assets/9291703/0bfd1aea-2ffb-459d-815d-08b536c21f40)

While there's a copy/paste button it would be more useful if it all fit on one page.  You can use the backslash `\` for bash commands to continue on the next line, so I think this might be better as:
```bash
 curl -XPOST "localhost:9200/_extensions/initialize" -H "Content-Type:application/json" --data '{ \
"name":"ad-extension", \
"uniqueId":"ad-extension", \
"hostAddress":"127.0.0.1", \
"port":"4532", \
"version":"1.0", \
"opensearchVersion":"3.0.0", \
"minimumCompatibleVersion":"3.0.0", \
"dependencies":[{"uniqueId":"test1","version":"2.0.0"},{"uniqueId":"test2","version":"3.0.0"}] \
}'
```

```

To **run OpenSearch from Gradle**, follow these steps:
- Run `./gradlew run` to start OpenSearch.
- Hit the below sample REST API to initialize an extension
```
 curl -XPOST "localhost:9200/_extensions/initialize" -H "Content-Type:application/json" --data '{"name":"ad-extension","uniqueId":"ad-extension","hostAddress":"127.0.0.1","port":"4532","version":"1.0","opensearchVersion":"3.0.0","minimumCompatibleVersion":"3.0.0","dependencies":[{"uniqueId":"test1","version":"2.0.0"},{"uniqueId":"test2","version":"3.0.0"}]}'
```

> `ExtensionsManager` discovers the extension listening on a predefined port and executes the TCP handshake protocol to establish a data transfer connection. Then OpenSearch sends a request to the OpenSearch SDK for Java and, upon acknowledgment, the extension responds with its name. This name is logged in the terminal where OpenSearch is running:

Reviewer: ```suggestion
In response to the REST `/initialize` request, `ExtensionsManager` discovers the extension listening on a predefined port and executes the TCP handshake protocol to establish a data transfer connection. Then OpenSearch sends a request to the OpenSearch SDK for Java and, upon acknowledgment, the extension responds with its name. This name is logged in the terminal where OpenSearch is running:
```

- Start a separate terminal and navigate to the directory where OpenSearch has been cloned using `cd OpenSearch`.
- Run `./gradlew assemble` to create a local distribution.
- Start OpenSearch using `./bin/opensearch`.
- Query the below sample REST API to initialize an extension

> ```

Reviewer 2: ```suggestion
```bash
```

curl -XPOST "localhost:9200/_extensions/initialize" -H "Content-Type:application/json" --data '{ \
"name":"ad-extension", \
"uniqueId":"ad-extension", \
"hostAddress":"127.0.0.1", \
"port":"4532", \
"version":"1.0", \
"opensearchVersion":"3.0.0", \
"minimumCompatibleVersion":"3.0.0", \
"dependencies":[{"uniqueId":"test1","version":"2.0.0"},{"uniqueId":"test2","version":"3.0.0"}] \
}'
```

To **run OpenSearch from Gradle**, follow these steps:
- Run `./gradlew run` to start OpenSearch.

> - Query the below sample REST API to initialize an extension

Reviewer 2: ```suggestion
- Send the following request to initialize an extension:
```

> ```

Reviewer 2: ```suggestion
```bash
```

- Start a separate terminal and navigate to the directory where OpenSearch has been cloned using `cd OpenSearch`.
- Run `./gradlew assemble` to create a local distribution.
- Start OpenSearch using `./bin/opensearch`.

> - Send the below sample REST API to initialize an extension

Reviewer 3: Do we want the default to be for AD or hello world?

```bash
curl -XPOST "localhost:9200/_extensions/initialize" -H "Content-Type:application/json" --data '{ \

> "name":"ad-extension", \

Reviewer: Yeah, either "hello-world" or "sample-extension" probably fits better here.
